homeassistant/components/light/mipow.py

```diff
@@ -0,0 +1,208 @@
+        """Return true if device is on."""
+        return self._state
+
+    @property
+    def rgb_color(self):
+        """Return the color property."""
+        return self._rgb
+
+    @property
+    def white_value(self):
+        """Return the white property."""
+        return self._white
+
+    @property
+    def brightness(self):
+        """Return the bright property."""
+        return self._brightness
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_MIPOW_LED
+
+    @property
+    def should_poll(self):
+        """Feel free to poll."""
+        return True
+
+    @property
+    def assumed_state(self):
+        """We can report the actual state."""
+        return False
+
+    def set_effect(self, red, green, blue, white, speed):
+        """Set the rgb state."""
+        return self._bulb.set_effect(red, green, blue, white, 0, speed)
+
+    def set_rgb(self, red, green, blue):
+        """Set the rgb state."""
+        return self._bulb.set_rgb(red, green, blue)
+
+    def set_white(self, white):
+        """Set the white state."""
+        return self._bulb.set_white(white)
+
+    def get_bright(self):
+        """Return Brightness."""
+        rgb = self._bulb.get_colour()
+        hsv = self.hsv_conv(rgb[0], rgb[1], rgb[2])
+        return (hsv[2]) * 255
+
+# pylint: disable=R0201
+    def rgb_conv(self, hue, sat, val):
+        """Used to convert hsv to rgb."""
+        import colorsys
+        rgb = colorsys.hsv_to_rgb(hue, sat, val)
+        return rgb
+
+# pylint: disable=R0201
+    def hsv_conv(self, re, gr, bl):
+        """Used to convert rgb to hsv."""
+        import colorsys
+        hsv = colorsys.rgb_to_hsv(re/255, gr/255, bl/255)
+        return hsv
+
+    def turn_on(self, **kwargs):
+        """Turn the specified light on."""
+        rgb = kwargs.get(ATTR_RGB_COLOR)
+        white = kwargs.get(ATTR_WHITE_VALUE)
+        bright = kwargs.get(ATTR_BRIGHTNESS)
+        flash = kwargs.get(ATTR_FLASH)
+        if self._state is False:
+            if rgb is None and bright is None and flash is None:
+                self._white = 255
+                white = 255
+                self._state = True
+                self._bulb.on()
+        self._state = True
+
+        if flash is not None:
+            if flash == FLASH_LONG:
+                self._speed = 2
+            elif flash == FLASH_SHORT:
+                self._speed = 1
+
+        if bright is not None:
+            rgbs = self._bulb.get_colour()
+            hsvs = self.hsv_conv(rgbs[0], rgbs[1], rgbs[2])
+            bri = bright / 255
+            rgbv = self.rgb_conv(hsvs[0], hsvs[1], bri)
+            respred = int(round(rgbv[0]*255))
+            respgreen = int(round(rgbv[1]*255))
+            respblue = int(round(rgbv[2]*255))
+            self._rgb = (respred, respgreen, respblue)
+            self._white = 0
+        elif rgb is not None:
+            self._white = 0
```

Comment:
Does the bulb not support simultaneous white / rgb values? Otherwise don't make these ifs exclusive, and only modify the relevant attribute, (i.e. rgb should not set white to zero).